packages/editor-ui/src/components/ExecutionsUsage.vue

```diff
@@ -0,0 +1,184 @@
+		<div v-if="!isTrialExpired && trialHasExecutionsLeft" :class="$style.usageText">
+			<i18n path="executionUsage.currentUsage">
+				<template #text>
+					<n8n-text size="xsmall" color="text-dark">
+						{{ locale.baseText('executionUsage.currentUsage.text') }}
+					</n8n-text>
+				</template>
+				<template #count>
+					<n8n-text size="xsmall" :bold="true" color="warning">
+						{{ daysLeftOnTrial }} {{ locale.baseText('executionUsage.currentUsage.count') }}
+					</n8n-text>
+				</template>
+			</i18n>
+		</div>
+		<div v-if="isTrialExpired" :class="$style.usageText">
+			<n8n-text size="xsmall" color="danger">
+				{{ locale.baseText('executionUsage.expired.text') }}
+			</n8n-text>
+		</div>
+		<div v-if="!trialHasExecutionsLeft" :class="$style.usageText">
+			<n8n-text size="xsmall">
+				{{ locale.baseText('executionUsage.ranOutOfExecutions.text') }}
+			</n8n-text>
+		</div>
+		<div v-if="!isTrialExpired" :class="$style.usageCounter">
+			<div :class="$style.progressBarSection">
+				<progress
+					:class="[
+						trialHasExecutionsLeft ? $style.progressBarSuccess : $style.progressBarDanger,
+						$style.progressBar,
+					]"
+					:value="currentExecutions"
+					:max="maxExecutions"
+				></progress>
+			</div>
+			<div :class="$style.executionsCountSection">
+				<n8n-text size="xsmall" :color="trialHasExecutionsLeft ? 'text-dark' : 'danger'">
+					{{ currentExecutions }}/{{ maxExecutions }}
+				</n8n-text>
+				<n8n-text size="xsmall" :color="trialHasExecutionsLeft ? 'text-dark' : 'danger'">{{
+					locale.baseText('executionUsage.label.executions')
+				}}</n8n-text>
+			</div>
+		</div>
+
+		<div :class="$style.upgradeButtonSection">
+			<n8n-button
+				:label="locale.baseText('executionUsage.button.upgrade')"
+				size="mini"
+				icon="gem"
+				type="success"
+				@click="onUpgradeClicked"
+			/>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { onMounted, ref, computed } from 'vue';
+import { i18n as locale } from '@/plugins/i18n';
+import type { PlanData } from '@/Interface';
+import { DateTime } from 'luxon';
+
+const currentPlan = ref<PlanData>({
+	planSpec: {
+		planId: 43039,
+		monthlyExecutionsLimit: 200,
+		activeWorkflowsLimit: 10,
+		credentialsLimit: 100,
+		isActive: false,
+		displayName: 'Trial',
+		metadata: {
+			version: 'v1',
+			group: 'trial',
+			slug: 'trial-1',
+			trial: {
+				length: 7,
+				gracePeriod: 3,
+			},
+		},
+	},
+	instance: {
+		createdAt: '2023-05-01T01:47:47Z',
+	},
+	usage: {
+		executions: 100,
+		activeWorkflows: 10,
+	},
+});
+
+const now = DateTime.utc();
+
+onMounted(async () => {});
+
+const daysLeftOnTrial = computed(() => {
+	const { days = 0 } = now.diff(getPlanStartingDate(), ['days']).toObject();
+	return Math.trunc(days);
+});
+
+const isTrialExpired = computed(() => {
+	const trialEndsAt = getPlanStartingDate()
+		.plus({ days: currentPlan.value.planSpec.metadata.trial.length })
+		.endOf('day');
```

Comment:
hmm why endOf? how do we ensure the backend and FE are consistent? considering hooks have to be backwards compatible, maybe we should pass the end date to the FE instead.. to have consistent source of truth for this and avoid any drift between FE and BE in the future..

Reply:
> hmm why endOf? 

Probably too many hours on the computer 😅

> how do we ensure the backend and FE are consistent? considering hooks have to be backwards compatible, maybe we should pass the end date to the FE instead.. to have consistent source of truth for this and avoid any drift between FE and BE in the future.

That is a very good idea. I will add it to the endpoint and mock it up here.